From 3834991c4c6eb76ffc26dc45634fdef3550db7ba Mon Sep 17 00:00:00 2001
From: Nancy Chen <Nancy.Chen@freescale.com>
Date: Fri, 15 Feb 2008 16:07:17 -0600
Subject: [PATCH] ENGR00066035 Enable regulator by default on mx31ADS and MX31_3stack

Description:
Enable regulator by default on mx31ADS and MX31_3stack

Signed-off-by: Nancy Chen <Nancy.Chen@freescale.com>
---
 arch/arm/configs/imx31_3stack_defconfig  |    7 ++++++-
 arch/arm/configs/imx31ads_defconfig      |    7 ++++++-
 include/asm-arm/arch-mxc/pmic_external.h |    7 +++++++
 3 files changed, 19 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx31_3stack_defconfig b/arch/arm/configs/imx31_3stack_defconfig
index a275900..a61c453 100644
--- a/arch/arm/configs/imx31_3stack_defconfig
+++ b/arch/arm/configs/imx31_3stack_defconfig
@@ -479,7 +479,12 @@ CONFIG_MTD_NAND_MXC_V2=y
 #
 # Voltage and Current regulators
 #
-# CONFIG_REGULATOR is not set
+CONFIG_REGULATOR_API=y
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_MC13783=y
+
+#
+#
 # CONFIG_PARPORT is not set
 # CONFIG_PNP is not set
 CONFIG_BLK_DEV=y
diff --git a/arch/arm/configs/imx31ads_defconfig b/arch/arm/configs/imx31ads_defconfig
index e954220..5e7cee3 100644
--- a/arch/arm/configs/imx31ads_defconfig
+++ b/arch/arm/configs/imx31ads_defconfig
@@ -556,7 +556,12 @@ CONFIG_MTD_NAND_MXC_V2=y
 #
 # Voltage and Current regulators
 #
-# CONFIG_REGULATOR is not set
+CONFIG_REGULATOR_API=y
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_MC13783=y
+
+#
+#
 # CONFIG_PARPORT is not set
 # CONFIG_PNP is not set
 CONFIG_BLK_DEV=y
diff --git a/include/asm-arm/arch-mxc/pmic_external.h b/include/asm-arm/arch-mxc/pmic_external.h
index 39cfe69..71e1e03 100644
--- a/include/asm-arm/arch-mxc/pmic_external.h
+++ b/include/asm-arm/arch-mxc/pmic_external.h
@@ -875,12 +875,19 @@ bool pmic_check_sensor(t_sensor sensor);
 */
 PMIC_STATUS pmic_get_sensors(t_sensor_bits * sensor_bits);
 
+#ifdef CONFIG_REGULATOR_MC13783
 /*!
  * This function is used to initialize the regulator for MC13783.
  *
  * @return      Returns 0.
  */
 int reg_mc13783_probe(void);
+#else
+static inline int reg_mc13783_probe(void)
+{
+	return 0;
+};
+#endif
 #endif				/* __KERNEL__ */
 
 #endif				/* __ASM_ARCH_MXC_PMIC_EXTERNAL_H__ */
-- 
1.5.4.4

